crates/blockchain-tree/src/blockchain_tree.rs

```diff
@@ -1160,29 +1160,43 @@ impl<DB: Database, EF: ExecutorFactory> BlockchainTree<DB, EF> {
         chain: Chain,
         recorder: &mut MakeCanonicalDurationsRecorder,
     ) -> RethResult<()> {
-        // Compute state root before opening write transaction.
-        let hashed_state = chain.state().hash_state_slow();
-        let (state_root, trie_updates) = chain
-            .state()
-            .state_root_calculator(
-                self.externals.provider_factory.provider()?.tx_ref(),
-                &hashed_state,
-            )
-            .root_with_updates()
-            .map_err(Into::<DatabaseError>::into)?;
-        let tip = chain.tip();
-        if state_root != tip.state_root {
-            return Err(RethError::Provider(ProviderError::StateRootMismatch(Box::new(
-                RootMismatch {
-                    root: GotExpected { got: state_root, expected: tip.state_root },
-                    block_number: tip.number,
-                    block_hash: tip.hash,
-                },
-            ))))
-        }
+        let (blocks, state, chain_trie_updates) = chain.into_inner();
+        let hashed_state = state.hash_state_slow();
```

Comment:
Should we call this only if there are no chain_trie_updates?

Reply:
Ah i see, theanswer is not. Maybe we can add those similar to how we added `chain_trie_updates` on the happy path, but that is for later.